### PR TITLE
Made the ImGui debug tools not disable the rest of the Editor.

### PR DESCRIPTION
**Pull request by santorac**
## What does this PR do?

Before, when you enabled the ImGui system (Home key) in edit mode, this would constrain the mouse to the 3D viewport and disable everything else in the Editor. It appears the only reason for this limitation was a bug where the engine's mouse coordinates were normalized to the entire window rather than being normalized to just the 3D viewport, which resulted in incorrect mouse coordinates being passed to ImGui. Enabling the ConstrainedAndHidden mode was a workaround that forced the engine's mouse coordinates to be normalized to the 3D viewport. I presume the reason the other Editor UI was being disabled was make users less frustrated about the fact that the cursor was constrained.

This limitation blocked our ability to create ImGui debug tools that benefit from interaction with entity components, because you couldn't interact with both at the same time.

I was able to fix the coordinate issues, allowing the removal of these workaround. Now you can turn on the ImGui system and seamlessly interact with Material Components at the same time.

Other changes I made:
- There was an issue where discrete mode would be unavailable (at least by default) in game mode if you previously opened ImGui in edit mode.
- I hid the Discrete menu item checkbox in edit mode because it's unnecessary there.

ImGuiInteraction

## How was this PR tested?

Windows only (I don't have access to other platforms).
Editor in edit mode. Discrete mode is not present. Mouse UI interaction is seamless.
Game launcher and Editor in game mode. Discrete mode is enabled by default. The Home key cycles between ImGui input and Game input. Discrete mode can be enabled to control both at the same time. Mouse contraints can be changed manually through the menus.

Signed-off-by: santorac <55155825+santorac@users.noreply.github.com>

### CODE REVIEW

**Review by hultonha — APPROVED** [review]
This looks like a great change! Awesome work improving the current workflow and +1 to all the helpful additional comments too.

Might be worth tapping @pollend or another Linux user to try out the change on a different platform to make sure there aren't any unintended consequences from the change on other platforms. Better to know sooner rather than later 😝 Otherwise looks good! 🙂

**Review by AMZN-daimini — APPROVED** [review]

**Review by AMZN-ScottR — APPROVED** [review]
Nice find

File: Code/Editor/EditorViewportWidget.cpp [hunk 1]
```diff
@@ -2295,16 +2295,11 @@ void EditorViewportWidget::EndUndoTransaction()
...
-
-    AzFramework::InputSystemCursorRequestBus::EventResult(
-        systemCursorState, AzFramework::InputDeviceMouse::Id, &AzFramework::InputSystemCursorRequests::GetSystemCursorState);
-
-    const bool systemCursorConstrained =
-        (systemCursorState == AzFramework::SystemCursorState::ConstrainedAndHidden ||
-         systemCursorState == AzFramework::SystemCursorState::ConstrainedAndVisible);
-
-    return systemCursorConstrained ? renderOverlayHWND() : nullptr;
+    // Even when the mouse cursor is not in a constrained mode, we still return the this viewport as the constraint window,
```
Comment (cgalvan) [inline Code/Editor/EditorViewportWidget.cpp:2298]:
```suggestion
    // Even when the mouse cursor is not in a constrained mode, we still return the viewport as the constraint window,
```

File: Code/Editor/MainWindow.cpp [hunk 2]
```diff
@@ -508,16 +508,6 @@ void MainWindow::Initialize()
     ActionOverrideRequestBus::Event(
         GetEntityContextId(), &ActionOverrideRequests::SetupActionOverrideHandler, this);
 
-    if (auto imGuiManager = AZ::Interface<ImGui::IImGuiManager>::Get())
-    {
-        auto handleImGuiStateChangeFn = [](bool enabled)
-        {
-            EditorWindowUIRequestBus::Broadcast(&EditorWindowUIRequests::SetEditorUiEnabled, enabled);
-        };
-        m_handleImGuiStateChangeHandler = ImGui::IImGuiManager::ImGuiSetEnabledEvent::Handler(handleImGuiStateChangeFn);
```
Comment (cgalvan) [inline Code/Editor/MainWindow.cpp:517]: Can also remove the `m_handleImGuiStateChangeHandler` member in `MainWindow` now that its unused, and I think the `ImGuiBus.h` include above

File: Gems/ImGui/Code/Source/LYCommonMenu/ImGuiLYCommonMenu.cpp [hunk 3]
```diff
@@ -513,6 +513,8 @@ namespace ImGui
                     }
 
                     // Discrete Input Mode
+                    // Not available in edit mode because inputs are discrete there anyway.
+                    if(!gEnv->IsEditor() || gEnv->IsEditorGameMode())
```
Comment (cgalvan) [inline Gems/ImGui/Code/Source/LYCommonMenu/ImGuiLYCommonMenu.cpp:517]:
```suggestion
                    if (!gEnv->IsEditor() || gEnv->IsEditorGameMode())
```

**Review by cgalvan — APPROVED** [review]

File: Code/Editor/MainWindow.cpp (hunk 2 above)
Comment (santorac) [reply Code/Editor/MainWindow.cpp:517]: Okay, I'll take care of that in my next PR.

**Pull request merged**